File: 2024/beginner/13-automat/solution/solution.py



if __name__ == "__main__":
    EXISTING_VALUES = [1, 2, 5, 10, 20, 50]
    EXISTING_VALUES.sort(reverse=True)

    C = int(input()) # coffee cost

    V = int(input())

    entered_sum = 0
    for i in range(V): entered_sum += int(input())

    to_return = entered_sum - C

    if to_return < 0:
        print("nelze vydat")
        exit()

    for val in EXISTING_VALUES:
        for i in range(to_return // val): print(val)
        to_return %= val

File: 2024/beginner/13-automat/solution/solution.cpp
#include <iostream>




int possible_values[] = {50,20,10,5,2,1};

int main(){
    int C, V;
    std::cin >> C >> V;

    int total_sum=0;
    for(int i=0; i < V; i++){
        int n;
        std::cin >> n;
        total_sum += n;
    }

    int to_return = total_sum-C;

    if(to_return < 0){
        std::cout << "nelze vydat\n";
        return 0;
    }

    for(int i=0; i<6; i++){
        for(int j=0; j < (int)(to_return/possible_values[i]); j++)
            std::cout << possible_values[i] << '\n';

        to_return %= possible_values[i];
    }


}